pkg/azuredisk/controllerserver.go

```diff
@@ -515,6 +515,9 @@ func (d *Driver) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsReques
 	if len(req.GetSnapshotId()) != 0 {
 		snapshot, err := d.getSnapshotByID(ctx, req.GetSnapshotId(), req.SourceVolumeId)
 		if err != nil {
+			if strings.Contains(err.Error(), "none-exist-id") {
```

Comment:
how do you know this const string `none-exist-id`, is it a const from somewhere?

Reply:
ok. It is from log. Deeply, it is the message from azure server. And I should use "ResourceNotFound" as identifier.